tajo-core/src/main/java/org/apache/tajo/engine/function/FunctionLoader.java

```diff
@@ -66,7 +80,48 @@
       }
     }
 
-    return map.values();
+    return map;
+  }
+
+  /**
+   * Load functions that are optionally defined by users.
+   *
+   * @param conf
+   * @param functionMap
+   * @return
+   * @throws IOException
+   */
+  public static Map<FunctionSignature, FunctionDesc> loadOptionalFunctions(TajoConf conf,
```

Comment:
In my opinion, 'user-defined function' would be a better word than 'optional function'